### ISSUE TITLE
`type[Any]` should allow access to any member that is not defined on `type` or `object` class

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#6552]
[This typing discussion thread](https://discuss.python.org/t/inconsistencies-between-type-and-type/37404) clarified that type checkers should allow access to members of `type[Any]` that are not explicitly declared in the `type` or `object` classes.

The following code snippet should produce no type errors:

```python
from typing import Any, Type

def func1(t1: Type, t2: Type[Any], t3: type, t4: type[Any]):
    reveal_type(t1.x)  # Should be `Unknown`
    reveal_type(t2.x)  # Should be `Any`
    reveal_type(t3.x)  # Should be `Unknown`
    reveal_type(t4.x)  # Should be `Any`
```

Currently, pyright emits four errors in this code, one for each line.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.338, which I just published. It will also be included in a future release of pylance.
